### DISCUSSION TITLE
Is it possible to use Grapevine in Windows Form apps?

### DISCUSSION THREAD

**Discussion by MSLight** [scottoffen/grapevine#141] [General]
I downloaded the source and Import [Grapeseed](https://github.com/scottoffen/grapevine/tree/main/src/Grapeseed) and [Grapevine](https://github.com/scottoffen/grapevine/tree/main/src/Grapevine) into my solution that contain my own Windows Form Project.

I wrote this codes:
```
private void Main_Load(object sender, EventArgs e)
{
  LoadApiServer();
}

private void LoadApiServer()
{
  using (var server = RestServerBuilder.UseDefaults().Build())
  {
    server.Start();
    MyLogBox.WriteLine("API Server Started.");
  }
}
```
This is MyResource Class:
```
namespace MyWinFormApplication
{
    [RestResource]
    public class MyResource
    {
        [RestRoute("Get", "/api/info")]
        public async Task Test(IHttpContext context)
        {
            await context.Response.SendResponseAsync("Info Api Called");
        }
    }
}
```

After run the Winform Application, server not starts and not listening on default port (localhost:1234) and I can't use it.

**Is it possible to use Grapevine in Windows Form apps?**

**Comment by scottoffen** [comment]
Yes, it is possible to use Grapevine in a Windows Form app. Your code is disposing of the server at the end of the method, so it won't be running anymore after that method call.
